### ISSUE TITLE
OOP approach to Bayesian EOQ module

### ISSUE THREAD

**Issue by jakorostami** [supplyseer-ai/supplyseer#21]
**Is your feature request related to a problem? Please describe.**
Rewrite the Bayesian EOQ module to be OOP

**Comment by yusufprasetyo25** [comment]
Hi Jako, I think taking this task will be a good start for me. Would you mind if I try it?

Is there anything I should be aware of before starting this project? Also, do you have any suggestions or references to help me get started?

**Comment by jakorostami** [comment]
Hi Yusuf!

There is already a PR on this issue.

See this: https://github.com/supplyseer-ai/supplyseer/pull/22 

However, I haven't approved it since a bit more work is needed on it. Try out that one and modify where modification is needed.

What do you think?

**Comment by yusufprasetyo25** [comment]
Ok, sure! Will take a look at it

**Comment by yusufprasetyo25** [comment]
Hi Jako. I checked the Bayesian EOQ that we are going turn to OOP.

I noticed that the `std` of the normal PDF is hardcoded to be 0.1 of the mean value.
![image](https://github.com/user-attachments/assets/a63a1a00-f4c3-42c9-b714-76d4b1d3bcce)

Should we turn it into a parameter instead? Would like to hear your thoughts about this. Thank you!

**Comment by jakorostami** [comment]
That is a very good idea. Have you adapted it?

**Comment by yusufprasetyo25** [comment]
Sure! I was in the process in doing so.

I also came up with more questions (please bear with me 🥲) when I rewrite it:
1. I looked your #22 and found that you already did most of the job, but you said that more work is needed. Would you elaborate more on things that need to be worked on so that I can deliver to your expectation?
2. The `min_least_probable` eoq should use minimal `d`, `a`, but maximal `h`, right? Since `h` is used as denominator. Same logic goes to `max_least_probable` eoq, should it use minimal `h` instead?

**Comment by jakorostami** [comment]
> Sure! I was in the process in doing so.
> 
> 
> 
> I also came up with more questions (please bear with me 🥲) when I rewrite it:
> 
> 1. I looked your #22 and found that you already did most of the job, but you said that more work is needed. Would you elaborate more on things that need to be worked on so that I can deliver to your expectation?
> 
> 2. The `min_least_probable` eoq should use minimal `d`, `a`, but maximal `h`, right? Since `h` is used as denominator. Same logic goes to `max_least_probable` eoq, should it use minimal `h` instead?
> 
> 

No worries.

You can refactor my PR code with your suggested improvements. Make sure it passed the tests.

Give me your solution and raise a PR so I can review it. We'll take it from there!

**Comment by yusufprasetyo25** [comment]
Understood!
I'll put my suggestions in the upcoming PR so that we can review it together 🙂

**Comment by yusufprasetyo25** [comment]
Hi Jako! I tried to commit but it says no `.pre-commit-config.yaml` file was found.
I wonder if I can take a look at how the yaml file should be.
![image](https://github.com/user-attachments/assets/a02625c7-bc3a-4114-aec8-65ff5aec5414)
Your help would be greatly appreciated. Thank you!

**Comment by jakorostami** [comment]
> Hi Jako! I tried to commit but it says no `.pre-commit-config.yaml` file was found. I wonder if I can take a look at how the yaml file should be. ![image](https://private-user-images.githubusercontent.com/88311761/397715461-a02625c7-bc3a-4114-aec8-65ff5aec5414.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQ3MDE0NTgsIm5iZiI6MTczNDcwMTE1OCwicGF0aCI6Ii84ODMxMTc2MS8zOTc3MTU0NjEtYTAyNjI1YzctYmMzYS00MTE0LWFlYzgtNjVmZjVhZWM1NDE0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjIwVDEzMjU1OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTVlMjJhZDI3Mzc0Y2I1OGQyMDkxOTQ0YjRlZDYyZjY5NDMwMGYwZTEyYWQ4OTE1ZjM3MzdhZmI1ZDE2OTIxNTEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.scA88FbgzRD5Jxe3I6rm3Q7g-UEGKEot1BMIi4k9hsM) Your help would be greatly appreciated. Thank you!

I see what the error is. My fault here since I haven't updated with the pre-commit based dev branch. I assume you have installed pre-commit.

In your working folder of supplyseer just do `pre-commit uninstall` and commit again!

**Comment by yusufprasetyo25** [comment]
Thanks! Already submitted the PR: #47 
Please kindly review it and inform me if there is something need to be fixed 😄

**Comment by jakorostami** [comment]
> Thanks! Already submitted the PR: #47 
> 
> Please kindly review it and inform me if there is something need to be fixed 😄 

Amazing and congrats. Are you ready for a heavy review? 😂

Will take the time later and review.

Again, good job.

**Comment by yusufprasetyo25** [comment]
Yes, please help me review it closely so that we can deliver our best result. Thank you!
